### DISCUSSION TITLE
support for repeated use with script and list of passwords

### DISCUSSION THREAD

**Discussion by JoshBeckmann** [GurpreetKang/BitwardenDecrypt#11] [General]
Thanks for making this tool. I have a bitwarden vault of mine I'm trying to decrypt. I'd like to be able to use this script to try a csv of my potential passwords. I'm a beginning programmer. Can you suggest a way to use your tool like this? I'd like to use windows, but I could use ubuntu if necessary. I reviewed your code, but I wasn't able to follow it easily. (My limitation- not yours.)

Thanks again, 
Josh

**Comment by GurpreetKang** [comment]
Can you elaborate a little more about what you are trying to do?  Do you have a CSV file containing only potential passwords? Currently BitwardenDecrypt is interactive (it prompts you for the password) but could be easily modified to input a password as a command line argument passed by another script. Though this would be a one-off for your purpose, as I don't see a good use case for it to be available for everyone.

You would need another script that:
- reads a potential password from csv file
- calls BitwardenDecrypt with potential password
- checks exit code
- if not 0, try next potential password


BitwardenDecrypt runs fine on Windows, you just need to have [Python 3](https://www.python.org/downloads/windows/) installed.

**Reply by JoshBeckmann** [reply]
Thanks for this. You understood my question correctly. I'm hoping to modify BitwardenDecrypt so it can input a password as a command line argument. I'll take a second look. Thanks again for the useful tool.

**Reply by GurpreetKang** [reply]
This is where the password is used (line 291):
https://github.com/GurpreetKang/BitwardenDecrypt/blob/d4fc46cbe89d541b734ec6c86854235063ca8215/BitwardenDecrypt.py#L290-L294

It needs to be provided to getBitwardenSecrets() as bytes, not a str.

**Reply by JoshBeckmann** [reply]
Appreciated. I'll see what I can do. Thanks again!

**Reply by JoshBeckmann** [reply]
I had good luck using autohotkey and programming a script that ran in the windows command line. This approach allowed me to open my own vault by testing about 10k combinations. Thanks for your help!

**Reply by GurpreetKang** [reply]
So you managed to decrypt your vault?

**Reply by JoshBeckmann** [reply]
I did. Your tool made it happen. Thanks!

**Reply by GurpreetKang** [reply]
Glad to hear it, now memorize that Master Password! 🙂
